src/Components/SearchBar/types.ts

```diff
@@ -1,6 +1,6 @@
 export interface SearchBarProps {
   placeHolder?: string;
   disabled?: boolean;
-  onSearch?: (query: string) => void;
+  onSearch: (query: string) => void;
   onClear?: () => void;
 }
```

Comment:
when you have a props object with optional arguments, they should come last, after the required ones.